dialog/fileitem.go

```diff
@@ -35,44 +41,67 @@ func (i *fileDialogItem) TappedSecondary(_ *fyne.PointEvent) {
 }
 
 func (i *fileDialogItem) CreateRenderer() fyne.WidgetRenderer {
-	img := canvas.NewImageFromResource(i.icon)
+	var img fyne.CanvasObject
+	if i.icon == nil {
+		mimeSub := i.mimeSubType
+		if len(mimeSub) > 5 {
+			mimeSub = mimeSub[:5]
+		}
+		img = NewFileIcon(i.mimeType, mimeSub)
+	} else {
+		img = canvas.NewImageFromResource(i.icon)
+	}
 	text := widget.NewLabelWithStyle(i.name, fyne.TextAlignCenter, fyne.TextStyle{})
 	text.Wrapping = fyne.TextTruncate
-	extText := canvas.NewText(i.ext, theme.BackgroundColor())
-	extText.Alignment = fyne.TextAlignCenter
-	extText.TextSize = theme.TextSize()
+
 	return &fileItemRenderer{item: i,
-		img: img, text: text, ext: extText, objects: []fyne.CanvasObject{img, text, extText}}
+		img: img, text: text, objects: []fyne.CanvasObject{img, text}}
 }
 
 func (i *fileDialogItem) isDirectory() bool {
 	return i.icon == theme.FolderIcon() || i.icon == theme.FolderOpenIcon()
 }
 
-func fileParts(path string) (name, ext string) {
+func fileParts(path string) (name, ext, mimeType, mimeSubType string) {
 	name = filepath.Base(path)
 	ext = filepath.Ext(name[1:])
 	name = name[:len(name)-len(ext)]
 
-	if len(ext) > 1 {
-		ext = ext[1:]
+	mimeTypeFull := mime.TypeByExtension(ext)
+	if mimeTypeFull == "" {
```

Comment:
How often does this happen?
If we open a directory of 1000 files then opening all the files to peek will likely be very slow.
Perhaps it would be better to leave out this more detailed check until we know it is required?

Reply:
Only on files with no extensions... so not that often in a directory with 1000s of files hopefully.